src/neuroconv/basedatainterface.py

```diff
@@ -46,6 +49,30 @@ def get_metadata(self):
 
         return metadata
 
+    def synchronize(self, starting_time: Optional[float] = None, timestamps: Optional[ArrayType] = None):
+        """
+        Synchronize all time references for this interface to the common time basis.
+
+        This can be either an offset shift to the 'starting_time' of a regularly sampled series,
+        or a vector of 'timestamps'.
+
+        Both are set in units seconds relative to the common 'session_start_time'.
+
+        Parameters
+        ----------
+        starting_time: float
+            The starting time for all temporal data in this interface.
+        timestamps: ArrayType
+            A full vector of timestamps all temporal data in this interface.
+        """
+        if starting_time is not None or timestamps is not None:
+            assert starting_time != timestamps, "Specify either 'starting_time' or 'timestamps', but not both!"
+
+        if starting_time is not None:
+            self.starting_time = starting_time
+        elif timestamps is not None:
+            self.timestamps = np.array(timestamps)
```

Comment:
I'd actually rather we throw a NotImplemented error here. Otherwise we'll have a seemingly functional `synchronize` method on all children that doesn't actually do anything. It would be find to define a method like `synchronization_validation` that always gets called first and put this code there.